pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/parser/Converter.groovy

```diff
@@ -64,8 +67,10 @@ public class Converter {
      * @throws ProcessingException If an error of high enough severity is detected in processing.
      */
     public static ProcessingReport validateJSONAgainstSchema(JSONObject origJson) throws ProcessingException {
-        JsonNode jsonNode = jacksonJSONFromJSONObject(origJson);
+        return validateJSONAgainstSchema(jacksonJSONFromJSONObject(origJson))
+    }
 
+    public static ProcessingReport validateJSONAgainstSchema(JsonNode jsonNode) throws ProcessingException {
         JsonSchema schema = ASTSchema.getJSONSchema();
 
         return schema.validate(jsonNode)
```

Comment:
although not really part of this PR, I've noticed some validation errors while converting from groovy are caught that are not when converting from json, could this be validated the same way here? One example is duplicate parallel names are allowed from json -> pipeline but not pipeline -> json